search/searcher/search_phrase.go

```diff
@@ -245,12 +247,15 @@ func (s *PhraseSearcher) checkCurrMustMatch(ctx *search.SearchContext) *search.D
 // a nil or empty TermLocationMap
 func (s *PhraseSearcher) checkCurrMustMatchField(ctx *search.SearchContext, tlm search.TermLocationMap) (
 	int, search.TermLocationMap) {
-	paths := findPhrasePaths(s.terms, tlm, 0)
+	if s.path == nil {
+		s.path = make(phrasePath, 0, 4)
```

Comment:
why 4?  we know the length of the slice of terms coming in, doesn't it make sense to use that?

Reply:
oh, good point